### PR TITLE
Lemonade forward

### PR DESCRIPTION
**Pull request by sbsmith86**
#### What's this PR do?

Organizes the JS for this "takeover" a little better, cleans it up and generalizes it so that it can be put anywhere. 

Starts to add some fun interaction stuff: 

repeat-signup-example

Code is still a little rough and the animation isn't exactly where I want it, but lets merge this in so we can continue to work. 
#### Relevant tickets

Addresses #6426

### CODE REVIEW

File: lib/themes/dosomething/paraneue_dosomething/js/signup/Signup.js [hunk 1]
```diff
@@ -22,53 +31,150 @@ const Signup = {
...
+      }).done(function(data) {
+        const signupId = data[0];
+
+        if (signupId) {
+          $this.val(_this.checkmark);
+          $this.css('background-color', _this.green);
+          $buttonContainer.html($this);
+        // @TODO - How should we handle campaigns folks are already signed up for.
+        // Do we even need to handle this case, ideally we would only show campaigns they
+        // are not signed up for.
```
Comment (chloealee) [inline lib/themes/dosomething/paraneue_dosomething/js/signup/Signup.js:94]: @sbsmith86 `dosomething_campaign_get_recommended_campaign_nids` function handles that [here](https://github.com/DoSomething/phoenix/blob/bb4fa06179e98a3b76cc14dd5a4791edfc8cf3e1/lib/modules/dosomething/dosomething_campaign/dosomething_campaign.module#L1028)!

Comment (sbsmith86) [inline lib/themes/dosomething/paraneue_dosomething/js/signup/Signup.js:94]: @chloealee greatness

**Comment by chloealee** [comment]
🎉  👍

**Comment by DFurnes** [comment]
Love that GIF. 🍋

**Comment by sbsmith86** [comment]
@DFurnes i love making gifs.

**Pull request merged**